ebean-core/src/test/java/org/tests/cache/TestQueryCache.java

```diff
@@ -299,4 +298,35 @@ public void findIds() {
     assertThat(sql).hasSize(1);
   }
 
+  @Test
+  public void findCountDifferentQueriesBit() {
+    DB.getDefault().getPluginApi().getServerCacheManager().clearAll();
```

Comment:
I've added this test, which discovered the initial problem, the missing `match` in `BitwiseExpression` and the I noticed, that it would be easy to produce hashCollisions if the magic number `92821` is used.

I also refactored that class to use `DB` instead of `Ebean`